pytorch_lightning/utilities/device_parser.py

```diff
@@ -241,10 +240,16 @@ def _check_data_type(device_ids: Any) -> None:
         MisconfigurationException:
             If ``device_ids`` of GPU/TPUs aren't ``int``, ``str``, sequence of ``int`` or ``None``
     """
-    if device_ids is not None and (
-        not isinstance(device_ids, (int, str, MutableSequence, tuple)) or isinstance(device_ids, bool)
-    ):
-        raise MisconfigurationException("Device ID's (GPU/TPU) must be int, string or sequence of ints or None.")
+    msg = "Device IDs (GPU/TPU) must be an int, a string, a sequence of ints or None, but you passed"
+
+    if device_ids is None:
+        return
+    elif isinstance(device_ids, (MutableSequence, tuple)):
+        for id_ in device_ids:
+            if type(id_) is not int:
+                raise MisconfigurationException(f"{msg} a sequence of {type(id_).__name__}.")
+    elif type(device_ids) not in [int, str]:
```

Comment:
```suggestion
    elif type(device_ids) not in (int, str):
```
super nit:

Reply:
"super nit", haha, love that @rohitgr7 :D